File: src/markoun/core/model/history.py
from pydantic import BaseModel, Field


class HistoryNodeResponse(BaseModel):
    id: str
    parent_id: str | None
    created_at: str
    sequence: int
    content_size: int
    author: str | None
    path: str
    message: str | None


class HistoryTreeResponse(BaseModel):
    note_id: str
    root_node_id: str | None
    default_revision_id: str | None
    nodes: list[HistoryNodeResponse] = Field(default_factory=list)


class HistoryRevisionResponse(BaseModel):
    revision_id: str
    content: str


class HistoryDeleteRequest(BaseModel):
    filepath: str
    revision_id: str


class HistoryPurgeResponse(BaseModel):
    deleted_histories: int

File: web/src/types/history.ts
export type HistoryAvailability = 'unknown' | 'enabled' | 'disabled'

export interface HistoryNode {
  id: string
  parent_id: string | null
  created_at: string
  sequence: number
  content_size: number
  author: string | null
  path: string
  message: string | null
}

export interface HistoryTree {
  note_id: string
  root_node_id: string | null
  default_revision_id: string | null
  nodes: HistoryNode[]
}

export interface HistoryRevision {
  revision_id: string
  content: string
}
